packages/arb-token-bridge-ui/src/components/TransactionHistory/TransactionStatusInfo.tsx

```diff
@@ -4,101 +4,154 @@
+import { Button } from '../common/Button'
+import { isTxClaimable, isTxPending } from './helpers'
+import { Transition } from '../common/Transition'
 
 export const TransactionStatusInfo = () => {
   const { address } = useAccount()
   const { openTransactionHistoryPanel } = useAppContextActions()
   const { transactions } = useTransactionHistory(address)
 
-  const { numWithdrawalsReadyToClaim, numRetryablesToRedeem } = useMemo(() => {
+  const {
+    numClaimableTransactions,
+    numRetryablesToRedeem,
+    numPendingTransactions
+  } = useMemo(() => {
     return transactions.reduce(
       (acc, tx) => {
         // standard bridge withdrawal
-        if (isWithdrawalReadyToClaim(tx)) {
-          acc.numWithdrawalsReadyToClaim += 1
+        if (isTxClaimable(tx)) {
+          acc.numClaimableTransactions += 1
         }
         // failed retryable
         if (isDepositReadyToRedeem(tx)) {
           acc.numRetryablesToRedeem += 1
         }
+        // all pending
+        if (isTxPending(tx)) {
+          acc.numPendingTransactions += 1
+        }
         return acc
       },
-      { numWithdrawalsReadyToClaim: 0, numRetryablesToRedeem: 0 }
+      {
+        numClaimableTransactions: 0,
+        numRetryablesToRedeem: 0,
+        numPendingTransactions: 0
+      }
     )
   }, [transactions])
 
-  // don't show this banner if user doesn't have anything to claim or redeem
-  if (numWithdrawalsReadyToClaim === 0 && numRetryablesToRedeem === 0)
+  const shouldShow = useMemo(() => {
+    return (
+      numClaimableTransactions > 0 ||
+      numRetryablesToRedeem > 0 ||
+      numPendingTransactions > 0
+    )
+  }, [numClaimableTransactions, numRetryablesToRedeem, numPendingTransactions])
+
+  const Content = useCallback(() => {
+    const numClaimableTransactionsString = `claim ${numClaimableTransactions} ${
+      numClaimableTransactions === 1 ? 'transaction' : 'transactions'
+    }`
+    const numRetryablesToRedeemString = `retry ${numRetryablesToRedeem} ${
+      numRetryablesToRedeem === 1 ? 'transaction' : 'transactions'
+    }`
+    const numPendingTransactionsString = `${numPendingTransactions} pending ${
+      numPendingTransactions === 1 ? 'transaction' : 'transactions'
+    }`
+
+    if (numClaimableTransactions > 0 && numRetryablesToRedeem > 0) {
+      return (
+        <div className="flex space-x-2">
+          <ExclamationTriangleIcon width={20} />
+          <span>
+            Time sensitive: You must{' '}
+            <span className="font-bold">{numRetryablesToRedeemString}</span> and{' '}
+            <span className="font-bold">{numClaimableTransactionsString}</span>{' '}
+          </span>
+        </div>
+      )
+    }
+
+    if (numRetryablesToRedeem > 0) {
+      return (
+        <div className="flex space-x-2">
+          <ExclamationTriangleIcon width={20} />
+          <span>
+            You must{' '}
+            <span className="font-bold">{numRetryablesToRedeemString}</span>{' '}
```

Comment:
what's the deal with the space at the end?

Reply:
Mistake, I'll remove